### PR TITLE
Indexed storage

### PR DESCRIPTION
**Pull request by shjortConcordium**
## Purpose

 - Index storage of identities/accounts/IDPs under the current network.
 - Reset selected identity/account when switching network.
 - Block changing network during identity issuance.

## Changes

The purpose +
 - Removed identity id, and net from identities/credentials

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

**Comment by orhoj** [comment]
The list of identity providers is not reset when switching network. It probably works because we update it on-demand, but perhaps it should then be cleared when switching network to avoid any cross-polution.

**Comment by orhoj** [comment]
When switching network we should default a selected account. For example I only have 1 account, and it might as well become the selected account when switching around instead of showing a blank account screen.

File: packages/browser-wallet/src/popup/store/settings.ts [hunk 1]
```diff
@@ -22,9 +23,13 @@ const storedNetworkConfigurationAtom = atomWithChromeStorage<NetworkConfiguratio
 );
 export const networkConfigurationAtom = atom<NetworkConfiguration, NetworkConfiguration>(
     (get) => get(storedNetworkConfigurationAtom),
-    (_, set, networkConfiguration) => {
+    (get, set, networkConfiguration) => {
         set(storedNetworkConfigurationAtom, networkConfiguration);
         popupMessageHandler.broadcast(EventType.ChainChanged, networkConfiguration.genesisHash);
+        const accounts = get(accountsAtom);
+        const identities = get(identitiesAtom);
+        set(selectedAccountAtom, accounts.length ? accounts[0] : undefined);
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/store/settings.ts:31]: I guess my previous comment is that this does not seem to work.

File: packages/browser-wallet/src/popup/pages/NetworkSettings/NetworkSettings.tsx [hunk 2]
```diff
@@ -78,8 +78,34 @@ function NetworkConfigurationComponent({ networkConfiguration }: { networkConfig
...
+            <Modal disableClose open={Boolean(pendingIdentity)}>
+                <div className="network-settings-page__pending-identity">
+                    <p>{t('pendingIdentity.description')}</p>
+                    <Button
+                        width="narrow"
+                        className="network-settings-page__pending-identity__button"
+                        onClick={() => nav(-1)}
+                    >
+                        {t('pendingIdentity.back')}
+                        go back
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/NetworkSettings/NetworkSettings.tsx:96]: Translate or remove?

File: packages/browser-wallet/src/popup/pages/NetworkSettings/NetworkSettings.tsx [hunk 3]
```diff
@@ -78,8 +78,34 @@ function NetworkConfigurationComponent({ networkConfiguration }: { networkConfig
...
+                    </Button>
+                    <Button
+                        faded
+                        width="narrow"
+                        className="network-settings-page__pending-identity__button"
+                        onClick={() => setPendingIdentity(undefined)}
+                    >
+                        {t('pendingIdentity.abort')}
+                        abort issuance
```
Comment (orhoj) [inline packages/browser-wallet/src/popup/pages/NetworkSettings/NetworkSettings.tsx:105]: Translate or remove?

**Review by orhoj — CHANGES_REQUESTED** [review]

File: packages/browser-wallet/src/background/startup.ts [hunk 4]
```diff
@@ -23,26 +23,28 @@ async function monitorAccountStatus() {
...
-                        if (response?.status === TransactionStatusEnum.Finalized) {
-                            const isSuccessful = Object.values(response?.outcomes || {}).some(
-                                (outcome) => outcome.result.outcome === 'success'
-                            );
-                            toUpdate.push({
-                                ...info,
-                                status: isSuccessful ? CreationStatus.Confirmed : CreationStatus.Rejected,
-                            });
+            if (network) {
+                const url = network?.jsonRpcUrl;
```
Comment (orhoj) [inline packages/browser-wallet/src/background/startup.ts:27]:
```suggestion
                const url = network.jsonRpcUrl;
```

File: packages/browser-wallet/src/shared/storage/access.ts [hunk 5]
```diff
@@ -44,28 +63,77 @@ const makeStorageAccessor = <V>(area: chrome.storage.AreaName, key: ChromeStorag
...
+    area: chrome.storage.AreaName,
+    key: ChromeStorageKey
+): IndexedStorageAccessor<Value> => {
+    const inner = makeStorageAccessor<Record<string, Value>>(area, key);
+    return {
+        get: (index: string) => inner.get().then((s) => (s ? s[index] : undefined)),
+        set: async (index: string, value: Value) => {
+            const current = await inner.get();
+            const updated = { ...current };
+            updated[index] = value;
```
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/shared/storage/access.ts:82]: I think that is the same :thinking:

File: packages/browser-wallet/src/popup/store/settings.ts (hunk 1 above)
Comment (shjortConcordium) [reply packages/browser-wallet/src/popup/store/settings.ts]: Yeah, it used the list from the previous net. I've changed this, so it uses the correct net.

File: packages/browser-wallet/src/shared/storage/access.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (shjortConcordium) [reply packages/browser-wallet/src/shared/storage/access.ts:82]: or at least this has the same behaviour as the normal accessor.

**Review by orhoj — APPROVED** [review]

**Pull request merged**